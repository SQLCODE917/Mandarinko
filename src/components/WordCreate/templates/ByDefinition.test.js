import React from 'react'
import { mount } from 'enzyme'
import {
  getSuggestions,
  renderSuggestion,
  getSuggestionValue
} from './ByDefinition'

const allWords = require('../../../../testdata/testVocabulary.json')

describe('By Definition', () => {
  describe('getSuggestions', () => {
    it('should return an empty set for no input', () => {
      const suggestions = getSuggestions({})('');
      expect(suggestions).toEqual([]);
    })

    it('should return words on definition match', () => {
      const suggestions = getSuggestions(allWords)('vi')
      const expected = [
        {
          "children": [
            1,
            2
          ],
          "definition": [
            "(n) view, position, proposition",
            "(v) advocate, stand for, maintain, hold"
          ],
          "derivation": "",
          "id": "0",
          "pronounciation": "zhu(3)zhang(1)",
          "siblings": [
            5
          ],
          "spelling": [
            {
              "language": "zh-Hans",
              "text": "主张"
            },
            {
              "language": "zh-Hant",
              "text": "-張"
            }
          ]
        }
      ]
      expect(suggestions).toEqual(expected)
    })

    it('should return an empty set on no match', () => {
      const suggestions = getSuggestions(allWords)('ayyyyyyyyyy')
      expect(suggestions).toEqual([])
    })
  })

  describe('renderSuggestion', () => {
    it('should render', () => {
      const suggestion = getSuggestions(allWords)('vi').pop()
      const wrapper = mount(
        <div>
          {renderSuggestion(suggestion)}
        </div>
      )
      expect(wrapper).toMatchSnapshot()
    }) 
  })

  describe('getSuggestionValue', () => {
    it('formats it as a string', () => {
      const suggestion = getSuggestions(allWords)('vi').pop()
      const value = getSuggestionValue(suggestion)
      expect(value).toEqual('[主张][-張]')
    })
  })
})
