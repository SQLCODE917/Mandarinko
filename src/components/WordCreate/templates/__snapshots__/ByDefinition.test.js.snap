// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`By Definition renderSuggestion should render 1`] = `
<div>
  <article>
    <section
      className="suggestionSpelling"
    >
      主张
      -張
    </section>
    <section>
      <ul
        className="suggestionDefinitionList"
      >
        <li
          className="suggestionDefinition"
          key="0"
        >
          (n) view, position, proposition
        </li>
        <li
          className="suggestionDefinition"
          key="1"
        >
          (v) advocate, stand for, maintain, hold
        </li>
      </ul>
    </section>
  </article>
</div>
`;
